### PR TITLE
CRUD methods for attributes

### PR DESCRIPTION
**Pull request by mylius**
Adds CRUD methods for attributes to the client.

### CODE REVIEW

File: hari_client/client/client.py [hunk 1]
```diff
@@ -1642,3 +1644,153 @@ def create_attributes(
...
+            The attribute with the given attribute_id
+
+        Raises:
+            APIException: If the request fails.
+        """
+        return self._request(
+            "GET",
+            f"/datasets/{dataset_id}/attributes/{attribute_id}",
+            params=self._pack(locals(), ignore=["dataset_id", "attribute_id"]),
+            success_response_item_model=models.Attribute,
```
Comment (katia-qm) [inline hari_client/client/client.py:1702]: Why does `get_attribute` return `model.Attribute`, but `get_attributeS` returns `list[models.AttributeResponse]`?

**Review by katia-qm — CHANGES_REQUESTED** [review]
Create a separate attribute is missing (not bulk).
What about attribute values/metadata? I'm not sure whether they are needed or not

Comparing all the fields is very overwhelming, so I have to trust you here;)
Did you test all of the methods, to make sure everything works and the parsing doesn't break?

File: hari_client/client/client.py (hunk 1 above)
Comment (mylius) [reply hari_client/client/client.py]: Oversight on my part.

**Comment by mylius** [comment]
> Create a separate attribute is missing (not bulk). What about attribute values/metadata? I'm not sure whether they are needed or not

Having people deal with values and metadata is probably a bit overwhelming, which is why we created the attribute endpoints, which wrap both of those.

> Comparing all the fields is very overwhelming, so I have to trust you here;) Did you test all of the methods, to make sure everything works and the parsing doesn't break?

I did do all CRUD operations at least once for one Attribute, but I didn't populate all fields and try all variations.

File: hari_client/client/client.py [hunk 2]
```diff
@@ -1631,3 +1633,231 @@ def create_attributes(
...
+        median: typing.Any | None = None,
+        variance: typing.Any | None = None,
+        standard_deviation: typing.Any | None = None,
+        range: typing.Any | None = None,
+        average_absolute_deviation: typing.Any | None = None,
+        cumulated_frequency: typing.Any | None = None,
+        frequency: dict[str, int] | None = None,
+        question: str | None = None,
+        archived: bool | None = None,
+        ml_predidictions: dict[str, float] | None = None,
```
Comment (katia-qm) [inline hari_client/client/client.py:1792]: ml_predi_di_ctions :) also in docstring

File: hari_client/client/client.py [hunk 3]
```diff
@@ -1631,3 +1633,231 @@ def create_attributes(
...
+            params=self._pack(locals(), ignore=["dataset_id", "attribute_id"]),
+            success_response_item_model=models.Attribute,
+        )
+
+    def update_attribute(
+        self,
+        dataset_id: str,
+        attribute_id: str,
+        name: str | None = None,
+        annotatable_id: str | None = None,
```
Comment (katia-qm) [inline hari_client/client/client.py:1769]: annotatable id seems to be required on the api, could you please double-check?

**Review by katia-qm — CHANGES_REQUESTED** [review]

File: hari_client/models/models.py [hunk 4]
```diff
@@ -921,6 +921,93 @@ def populate_derived_field(cls, values):
         return values
 
 
+class Attribute(BaseModel):
+    id: str = pydantic.Field(title="ID")
+    name: str
```
[the remark this answers is not in the document]
Comment (mylius) [reply hari_client/models/models.py:926]: Not really, but I also don't think it is currently actively used.

**Review by katia-qm — APPROVED** [review]
Amazing, thanks a lot!

**Pull request merged**